fastchat/model/compression.py

```diff
@@ -102,26 +102,42 @@ def apply_compressed_weight(module, compressed_state_dict, target_device, prefix
 
 def load_compress_model(model_path, device, torch_dtype, use_fast, revision="main"):
     # partially load model
-    tokenizer = AutoTokenizer.from_pretrained(
-        model_path, use_fast=use_fast, revision=revision
-    )
-
+    # `use_fast=True`` is not supported for some models.
+    try:
+        tokenizer = AutoTokenizer.from_pretrained(
+        model_path, use_fast=use_fast, revision=revision,trust_remote_code=True)
+    
+    except Exception as e:
+        tokenizer = AutoTokenizer.from_pretrained(
+        model_path, use_fast=False, revision=revision,trust_remote_code=True)
     with init_empty_weights():
+        # `trust_remote_code` should be set as `True` for both AutoConfig and AutoModel
         config = AutoConfig.from_pretrained(
             model_path,
             low_cpu_mem_usage=True,
             torch_dtype=torch_dtype,
             revision=revision,
+            trust_remote_code=True
         )
-        model = AutoModelForCausalLM.from_config(config)
+        # some models are loaded by AutoModel but not AutoModelForCausalLM, such as chatglm, chatglm2
+        try:
+            model = AutoModelForCausalLM.from_config(config,trust_remote_code=True)
+        except Exception as e:
+            model = AutoModel.from_config(config,trust_remote_code=True)
         linear_weights = get_compressed_list(model)
 
     if os.path.exists(model_path):
         # `model_path` is a local folder
         base_pattern = os.path.join(model_path, "pytorch_model*.bin")
     else:
         # `model_path` is a cached Hugging Face repo
-        model_path = snapshot_download(model_path, revision=revision)
+        # We don't necessarily need to download the model' repo again if there is a cache. So check the default huggingface cache first.
```

Comment:
Does `snapshot_download` already check the cache?

Reply:
At least in my case, the answer is no. It would raise `LocalEntryNotFoundError` if I use `snapshot_download` directly because I can't access huggingface hub. So I suppose this commit would make it more robust, even if one can't access huggingface hub.
![image](https://github.com/lm-sys/FastChat/assets/22924096/1d6cbc43-33ef-4e7c-8b4b-3f6ae0b922a4)